platform/ios/PolyPodApp/PolyPod/HomeScreenView.swift

```diff
@@ -0,0 +1,653 @@
+    }
+}
+
+struct SmallCardView: View {
+    @Environment(\.homeScreenTileSizes) var sizes
+    @Environment(\.homeScreenFeatureSelected) var onFeatureSelected
+    
+    let card: Card
+    private let foregroundColor: Color
+    
+    init(card: Card) {
+        self.card = card
+        self.foregroundColor = card.backgroundColor.isLight ? .black : .white
+    }
+    
+    var body: some View {
+        VStack(alignment: .center) {
+            Image(uiImage: card.image)
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+            Spacer()
+            Text(card.title)
+                .foregroundColor(foregroundColor)
+                .font(HomeScreenConstants.SmallTile.title.font)
+                .multilineTextAlignment(HomeScreenConstants.SmallTile.title.alignment)
+        }
+        .padding([.leading, .trailing, .bottom], HomeScreenConstants.SmallTile.otherPadding)
+        .padding([.top], HomeScreenConstants.SmallTile.topPadding)
+        .frame(width: sizes.smallTileWidth, height: sizes.smallTileWidth)
+        .background(card.backgroundColor)
+        .cornerRadius(HomeScreenConstants.Tile.cornerRadius)
+        .onTapGesture {
+            onFeatureSelected(card.id)
+        }
+    }
+}
+
+struct FooterView: View {
+    let model: FooterViewModel
+    var openLearnMoreAction: () -> Void = { }
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: HomeScreenConstants.Footer.verticalSpacing) {
+            Text(model.title)
+                .font(HomeScreenConstants.Footer.title.font)
+                .multilineTextAlignment(HomeScreenConstants.Footer.title.alignment)
+            Text(model.description)
+                .font(HomeScreenConstants.Footer.description.font)
+                .multilineTextAlignment(HomeScreenConstants.Footer.description.alignment)
+            Image(model.imageName)
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .frame(alignment: .center)
+            Button(model.buttonTitle) {
+                openLearnMoreAction()
+            }
+            .font(HomeScreenConstants.Footer.Button.title.font)
+            .multilineTextAlignment(HomeScreenConstants.Footer.Button.title.alignment)
+            .padding()
+            .frame(maxWidth: .infinity, alignment: .center)
+            .foregroundColor(model.buttonBackgroundColor.isLight ? .black : .white)
+            .background(model.buttonBackgroundColor)
+            .cornerRadius(HomeScreenConstants.Tile.cornerRadius)
+            
+        }
+        .padding(HomeScreenConstants.Footer.padding)
+        .background(model.backgroundColor)
+        .cornerRadius(HomeScreenConstants.Tile.cornerRadius)
+    }
+}
+
+// MARK: - Preview
+
+struct HomeScreenView_Previews: PreviewProvider {
+    static let categories: [HomeScreenSectionModel] = [
+        .init(title: "Your Data",
+              cards: [
+                .init(id: UUID().uuidString,
```

Comment:
This is literally boilerplate, so I guess it's OK for the time being; still, DRYing it to an array or dictionary to the variable aspects of it might save a few lines of this file that has a whole lot of them.

Reply:
This is done this way so one can easily modify the content of a particular card at a particular position for the means of testing.